src/test/java/org/fogbowcloud/manager/core/intercomponent/xmpp/handlers/RemoteDeleteOrderRequestHandlerTest.java

```diff
@@ -0,0 +1,136 @@
+
+
+    private RemoteDeleteOrderRequestHandler remoteDeleteOrderRequestHandler;
+    private PacketSender packetSender;
+
+    private Order order;
+    private RemoteFacade remoteFacade;
+
+    @Before
+    public void setUp() {
+        this.remoteDeleteOrderRequestHandler = new RemoteDeleteOrderRequestHandler();
+
+        this.packetSender = Mockito.mock(PacketSender.class);
+
+        PowerMockito.mockStatic(PacketSenderHolder.class);
+        BDDMockito.given(PacketSenderHolder.getPacketSender()).willReturn(this.packetSender);
+
+        this.remoteFacade = Mockito.mock(RemoteFacade.class);
+
+        PowerMockito.mockStatic(RemoteFacade.class);
+        BDDMockito.given(RemoteFacade.getInstance()).willReturn(this.remoteFacade);
+    }
+
+    // test case: When call the handle method passing a valid IQ object, it must create an OK result IQ and return it.
+    @Test
+    public void testHandleWithValidIQ() throws FogbowManagerException, UnexpectedException {
+        //set up
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put("user-name", "fogbow");
+
+        FederationUser federationUser = new FederationUser("fake-id", attributes);
+
+        this.order = new ComputeOrder(federationUser, "requestingMember", "providingmember",
+                1, 2, 3, "imageId", null, "publicKey", new ArrayList<>());
+
+        Mockito.doNothing().when(this.remoteFacade).activateOrder(Mockito.eq(this.order));
+
+        IQ iq = createIq();
+
+        // exercise
+        IQ result = this.remoteDeleteOrderRequestHandler.handle(iq);
+
+        //verify
+        String orderId = order.getId();
+        String orderProvidingMember = order.getProvidingMember();
+        String expected = String.format(TAG_RESULT_IQ, orderId, orderProvidingMember);
+        Assert.assertEquals(expected, result.toString());
+    }
+
+    // test case: When an Exception occurs, the handle method must return a response error.
+    @Test
+    public void testHandleWhenThrowsException() throws FogbowManagerException, UnexpectedException {
+        //set up
+        this.order = new ComputeOrder(null, "requestingMember", "providingmember",
+                1, 2, 3, "imageId", null, "publicKey", new ArrayList<>());
+
+        Mockito.doThrow(new FogbowManagerException()).when(this.remoteFacade).deleteOrder(Mockito.anyString(),
+                Mockito.any(FederationUser.class), Mockito.any(ResourceType.class));
+
+        IQ iq = createIq();
+
+        // exercise
+        IQ result = this.remoteDeleteOrderRequestHandler.handle(iq);
+
+        //verify
+        Mockito.verify(this.remoteFacade, Mockito.times(1)).deleteOrder(Mockito.anyString(),
+                Mockito.any(FederationUser.class), Mockito.any(ResourceType.class));
+
+        String orderId = order.getId();
+        String orderProvidingMember = order.getProvidingMember();
+        String expected = String.format(TAG_RESULT_ERRO, orderId, orderProvidingMember);
```

Comment:
I had the feeling we are missing some information about the exception in the response, aren't we?